pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatPatternSpec.java

```diff
@@ -81,6 +82,23 @@ public DateTimeFormatter getDateTimeFormatter() {
     return _dateTimeFormatter;
   }
 
+  /**
+   * Validates the sdf pattern
+   */
+  public static void validateFormat(@Nonnull String sdfPatternWithTz) {
```

Comment:
Let's extract a method to return the `DateTimeFormatter` from `sdfPatternWithTz`, which can be shared in both the constructor and the validation

Reply:
Oh, I see your point. Nvm then.
One small optimization we can do here is to only validate the time zone if it is specified, and skip the `withZone(dateTimeZone)` which will just create a separate formatter